src/configuration/validate_configuration.ts

```diff
@@ -0,0 +1,9 @@
+import { IConfiguration } from './types'
+
+export function validateConfiguration(configuration: IConfiguration): void {
```

Comment:
Once all configuration is loaded and merged together, here we validate if any options are contradictory - schema conformance is already checked earlier.